Create a Pattern for `Footer 1` Design

danielwrobert opened the pull request:
<!-- Start by describing the changes made in this Pull Request, and the reason for such changes. -->
WIP: Create a pattern for `Footer 1`.

<!-- Reference any related issues or PRs here -->

Fixes #8952 

<!-- Don't forget to update the title with something descriptive. -->
<!-- If you can, add the appropriate labels -->

#### Accessibility

<!-- If you've changed or added any interactions, check off the appropriate items below. You can delete any that don't apply. Use this space to elaborate on anything if needed. -->

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

### Screenshots

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

| Before | After |
| ------ | ----- |
|        |       |

### Testing

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

1.
2.
3.

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested by users (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

### WooCommerce Visibility

<!-- Check this [this doc](../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WC core or not (part of the feature plugin or experimental)-->

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Performance Impact

<!-- Please document any known performance impact (positive or negative) here. If negative, provide some rationale for why this is an okay tradeoff or how this will be addressed. -->

### Changelog

> Add suggested changelog entry here.

danielwrobert committed:
Add initial PHP pattern file with markup.

albarin commented on patterns/footer-1.php:
```diff
@@ -0,0 +1,44 @@
+<div class="wp-block-group" style="padding-top:0;padding-right:var(--wp--preset--spacing--40);padding-bottom:0;padding-left:var(--wp--preset--spacing--40)"><!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+<div class="wp-block-group"><!-- wp:site-logo {"style":{"color":{"duotone":"var:preset|duotone|dark-grayscale"}}} /-->
+
+<!-- wp:navigation {"ref":242} /--></div>
```
We need to make sure we don't include any ids or refs in patterns, they will not exist in other stores.
Here we can just use `<!-- wp:navigation /-->`.

danielwrobert replied:
Addressed in 4b97232.

albarin commented on patterns/footer-1.php:
```diff
@@ -0,0 +1,44 @@
+<!-- wp:group {"tagName":"footer","layout":{"type":"default"}} -->
+<footer class="wp-block-group"><!-- wp:group {"style":{"spacing":{"blockGap":"0","padding":{"top":"0","right":"var:preset|spacing|40","bottom":"0","left":"var:preset|spacing|40"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"space-between"}} -->
+<div class="wp-block-group" style="padding-top:0;padding-right:var(--wp--preset--spacing--40);padding-bottom:0;padding-left:var(--wp--preset--spacing--40)"><!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap"}} -->
+<div class="wp-block-group"><!-- wp:site-logo {"style":{"color":{"duotone":"var:preset|duotone|dark-grayscale"}}} /-->
```
I would remove any styling and just use `<!-- wp:site-logo /-->`.

danielwrobert replied:
Addressed in 7168d70.

albarin commented on patterns/footer-1.php:
```diff
@@ -0,0 +1,44 @@
+?>
+
+<!-- wp:group {"tagName":"footer","layout":{"type":"default"}} -->
+<footer class="wp-block-group"><!-- wp:group {"style":{"spacing":{"blockGap":"0","padding":{"top":"0","right":"var:preset|spacing|40","bottom":"0","left":"var:preset|spacing|40"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"space-between"}} -->
```
I think the footer should probably have full width: `<!-- wp:group {"tagName":"footer","layout":{"type":"default"}} -->` and `<footer class="wp-block-group alignfull">` similar to the other footers we already have, like [footer-simple.php](https://github.com/woocommerce/woocommerce-blocks/blob/6fe5fbb887527242dab9ac7675ea2e54dc365dd8/patterns/footer-simple.php#L10).

danielwrobert replied:
Addressed in 16b7481.

albarin reviewed (changes requested):
Thanks for working on this 👏 
I left some comments with small tweaks. 
Would be nice also to improve the indentation, it makes it much better to read and spot mistakes (similar to this [footer-simple.php](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/patterns/footer-simple.php)).

For patterns, in general, what we also need to make sure is that they look good on different screen sizes and different theme colors (dark, light, etc.)

albarin commented on patterns/footer-1.php:
```diff
@@ -0,0 +1,44 @@
+<?php
+/**
+ * Title: WooCommerce Footer 1
```
Could you check with @shaunandrews how can we improve the title?

danielwrobert committed:
Update alignment for improved readability.

danielwrobert committed:
Remove unneeded ref on navigation block.

danielwrobert committed:
Remove unneeded styling on Site Logo block.

danielwrobert committed:
Add full-width alignment to pattern as default.

danielwrobert commented:
@albarin I've updated all of the items in your feedback and noted accordingly. I've also updated the indentation for the pattern itself. Thank you for your review!

That being said, I'm going to go ahead and close this PR for now, due to the updates in our designs and plans.
